AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin.md

```diff
@@ -1,68 +1,181 @@
+
+A comparison between the first level of results and the second level of results
+
+**First level of results**
+```
+query getPost($input: GetPostInput!) {
+  getPost(input: $input) {
+    id
+  }
+}
+```
+
+**Second level of results**
+```
+query getPost($input: GetPostInput!) {
+  getPost(input: $input) {
+    id
+    comments {
+      items {
+        id
+      }
+      nextToken
+    }
+  }
+}
+```
+- `GraphQLResponseDecoder` decodes the `post` response data to a mutable `JSONValue` object as an intermediate step.
+```json
+{
+  "id": "[POST_ID]"
 }
 ```
-- The selection set can contain multiple levels of data (the first level being the post, and second level being the list of comments), however `get(Post.self, byId: id)` controls this and only creates a selection set containng the first level to be retrieved. This provides a scalable approach to retrieve models with an "not loaded" association, and allow the developer to lazy load the assocations later.
-- `GraphQLResponseDecoder` will decode the "post" response post data to a `JSONValue` object as an intermediate step to allow modifications to the data.
-- The object is analyzed according to the model schema, association data (post.id and "post" field name) is stored at "comments" by creating and storing `AppSyncModelMetadata` at the "comments" key.
-- The object is then serialized and then decoded to the post instance. The `Post` decoder will instantiate the fields as normally while the "comments" field will be decoded by the `List<Comment>` that delegates its logic to the `AppSyncModelListDecoder`. 
+- `GraphQLResponseDecoder ` analyzes the object's model schema. It stores association data (e.g., `post.id` and `"post"` field name) as an `AppSyncModelMetadata` in the instance `JSONValue`'s `"comments"` key.
+```json
+{
+  "id": "[POST_ID]",
+  "comments": {
+    "appSyncAssociatedId": "[POST_ID]",
+    "appSyncAssociatedField": "post"
+  }
+}
+```
+- `GraphQLResponseDecoder` serializes object and decodes it to a `Post` instance as described above. The `Post` model-specific decoder instantiates scalar fields normally, while decoding the `comments` field into a `List<Comment>` that delegates its logic to a `AppSyncListDecoder`.
 - `AppSyncListDecoder` checks that the data can be successfully decoded to an `AppSyncModelMetadata` and stores this information in an `AppSyncListProvider` when instantiating a "not loaded" list with association data.
-The developer has the post instance and can lazy load the comments upon accessing it.
+
+The developer now has the `Post` instance and can either explicitly load the comments or lazy load the comments upon accessing iterator methods
+
+**Explicit load** 
+```swift
+if let comments = post.comments {
+  comments.fetch { fetchResults in
```

Comment:
very much a nit, but `fetch` is a Void return type right, where success just indicates that the results are now available in the `List`? If so, let's rename this `fetchResult`, and change the "print" statement below to something like `"fetch completed, list data is now loaded into memory"` or similar.

If in fact the `fetchResults` include data, do something with them in the example to make that clear.

Reply:
yup this returns void to avoid the customer confusion, will update to `result` since this is a Result type and update print statement